metrics/Proposals.md

```diff
@@ -0,0 +1,124 @@
+# Proposals
+
+New proposals for changes to the source code, during a certain period.
+
+## Description
+
+When a project uses code review processes, changes are not directly
+submitted to the code base, but are first proposed for discussion
+as "proposals for change to the source code".
+Each of these proposals are intended to be reviewed by other developers,
+who may suggest improvements that will lead to the original proposers
+sending new versions of their propoposals, until reviews are
+positive, and the code is accepted, or until it is decided that
+the proposal is declined.
+
+For example, "proposals" correspond to "pull requests" in the case of GitHub,
+to "merge requests" in the case of GitLab, and to "code reviews"
+or in some contexts "changesets" in the case of Gerrit.
+
+### Parameters
+
+Mandatory:
+
+* Period of time. Start and finish date of the period. Default: forever.
+
+    Period during which proposals are considered.
+    
+* Criteria for source code. Algorithm. Default: all files are source code.
+
+    If we are focused on source code, we need a criteria for deciding
+    whether a file is a part of the source code or not.
+    
+### Aggregators
+
+Usual aggregators are:
+
+* Count. Total number of proposals during the period.
+
+## Specific description: GitHub
+
+In the case of GitHub, a proposal is defined as a "pull request",
+as long as it proposes changes to source code files.
+
+The date of the proposal can be defined (for considering it in a period or not)
+as the date in which the pull request was submitted.
+
+### GitHub parameters
+
+None.
+
+## Specific description: GitLab
+
+In the case of GitLab, a proposal is defined as a "merge request",
+as long as it proposes changes to source code files.
+
+The date of the proposal can be defined (for considering it in a period or not)
+as the date in which the merge request was submitted.
+
+### GitLab parameters
+
+None.
+
+## Specific description: Gerrit
```

Comment:
There is an existing specific description for GHTorrent I think we should integrate. 

###  GHTorrent

```SQL
SELECT count(distinct pull_request_id) as num_still_open, projects.name as project_name, projects.url as url
FROM msr14.pull_request_history
    join pull_requests on pull_request_history.pull_request_id = pull_requests.id
    join projects on pull_requests.base_repo_id = projects.id
where pull_request_id not in
    (SELECT pull_request_id
    FROM msr14.pull_request_history
    where action = 'closed')
group by projects.id
```

Reply:
Reading the SQL, this seems like a different metric. the filter is for `pull_request_id not in ... where action = 'closed'`, which I thin is therefore counting pull requests that are still not closed. The metric defined here is for pull requests initiated during a certain period. Besides, I don't a filter on time.

This said, I have two more general concerns:

* GHTorrent is not exactly a data source. GitHub is, GHTorrent is providing secondary information on what the original data source provides. From this point of view, the GHTorrent database is more like the GrimoireLab database: it retrieves data from the original data source, and organizes it in some specific way.
* This part of the specification was not (at least in my mind) to provide reference implementations. For that we have an specific section, below. This was for discussing specific peculiarities of the data source, and in this case, that is done for the three data sources that we're proposing for change proposals: GitHub pull requests, GitLab merge requests, and Gerrit changesets.

Maybe we could discuss, from a more general point of view, if we want to change the format for the description of the metric....

Reply:
I see your point too. Let's discuss in a moment, during our meeting...